src/components/CoachProtectedRoute.jsx

```diff
@@ -0,0 +1,19 @@
+import React from "react"
+import { Navigate, Outlet } from "react-router-dom"
+import { useAuth } from "../context/AuthContext"
+import { toast } from "react-toastify"
+
```

Comment:
This is an intermediate PR.

We have to define the logic to handle the `user_type` authorization.